Explain validity check errors (#4476)
qgis/QGIS-Documentation: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst

information ("message" for the **error** layer, "FID" and "_errors" for the
**invalid** layer and  only "FID" for the **valid** layer):

> .. warning:: Is mandatory to have activated the option **Stop algorithm execution when a geometry is invalid**  in :menuselection:`Settings --> Options --> Processing --> General --> Invalid Feature`

Reviewer: ```suggestion
.. warning:: It's mandatory to have activated the option **Stop algorithm execution when a geometry is invalid**  in :menuselection:`Settings --> Options --> Processing --> General --> Invalid Feature`
```

Reviewer: mandatory to activate?

Reviewer: I'm thinking quickly, is-it not the reverse?

Author: I @Gustry in default installation "Stop algorithm execution when geometry is invalid" is true but most of the times users deactivate to have more "performance" because this option ignores invalid geometries. But then when they want to check again and forget to change to default behaviour the result of the algorithm is always empty results assuming that everything is ok. I can check if this behavior is continuing to happen. I notice this when I explain to my colleagues. 
What do you think? We keep with this or remove this warning?

information ("message" for the **error** layer, "FID" and "_errors" for the
**invalid** layer and  only "FID" for the **valid** layer):

.. warning::
   It is mandatory to have activated the option
   **Stop algorithm execution when a geometry is invalid**  in
   :menuselection:`Settings --> Options --> Processing --> General --> Invalid Feature`

>    otherwise the outputs will be empty.

Reviewer 2: disclaimer: I did not run the alg.
If I'm correct, here you are running the "check validity" alg so if you ask to stop when a geometry is invalid, then you stop the alg at the first error encountered then you get no more than one error in your output. Which would not reflect your dataset since some features have not been processed. Should it be "it's mandatory to **NOT** have activated..."?

Author: I can confirm that works with different options. Sorry for the noise. I tested again. I don't know why had this error in the past. For this reason, don't make sense to have this warning. I will remove it

Reviewer 2: But if you activate **Stop algorithm execution when a geometry is invalid**, you confirm that it runs till the end of the layer?

>              otherwise the outputs will be empty.

Reviewer 3: The suggested indentation change is for making travis happy.

>              otherwise the outputs will be empty.

Author: @havatv I have this issue in Windows OS users. let me test again.

>              otherwise the outputs will be empty.

Reviewer 3: This is an RST indentation issue, so it should not depend on the OS/platform. I suggest that you just commit my suggestion and see if that is enough to make Travis happy.

The attribute table of each generated vector layer will contain some additional

> information (numbers of error found and type of error):

Reviewer 2: ```suggestion
information (number of errors found and types of error):
```
?

      - The number of valid geometries.

Type of error messages and their meanings
.........................................

.. list-table:: If the GEOS method is used the following error messages can occur: 
   :widths: 30 30 40
   :header-rows: 1

   * - Message error
     - Explanation
     - Example

   * - Repeated point
     - This error happens when a given vertex is repeated.
     - .. figure:: img/geos_rep_point.png
          :align: center

   * - Ring self-intersection

>      - This error happens when a geometry touches itself and generate a ring.

Reviewer 2: ```suggestion
     - This error happens when a geometry touches itself and generates a ring.
```

      - The number of valid geometries.

Type of error messages and their meanings
.........................................

If the GEOS method is used the following error messages can occur:

+--------------------------------------+----------------------------------------------+-----------------------------------------+
| Message error                        | Explanation                                  | Example                                 |

> +--------------------------------------+----------------------------------------------+-----------------------------------------+

Reviewer 3: To indicate the header row.  But I still suggest that you try to use
`.. list-table::`

> +--------------------------------------+----------------------------------------------+-----------------------------------------+

Author: @havatv  I agree with what you say. Merge this and add the rest. Let me put in table style that you suggest. I will try to do this at the end o the day. Then we can merge this 1st PR.

> +--------------------------------------+----------------------------------------------+-----------------------------------------+

Reviewer 3: @jonnyforestGIS, it is quite easy for me to fix the list-table stuff (have been working a lot with that recently). So if you don't mind, I can do that now, so that you can concentrate on the more important stuff.
